### PR TITLE
config_database: various features/formatting fixes

### PR DESCRIPTION
**Pull request by alltilla**
### This PR has 3 new features and formatting fixes

---
`config_database: add option to list contexts and drivers`
New usage:
 * `./config_database.py`: print contexts
 * `./config_database.py -c CONTEXT`: print drivers of CONTEXT
 * `./config_database.py -c CONTEXT -d DRIVER`: print options of DRIVER
---
`config_database: support driver name aliases`
From now on the `pipe` `source` will be accessible as:
 * `config_database.py -c source -d pipe`
 * `config_database.py -c source -d fifo`

instead of only:
 * `config_database.py -c source -d fifo/pipe`
---
Change this:
```
tls()
tls(
  ca-dir(<string>)
  peer-verify(<string>)
  peer-verify(<yesno>)
)
```

to this:
```
tls(
  <empty>
  ca-dir(<string>)
  peer-verify(<string> | <yesno>)
)
```

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: contrib/config_option_database/config_database.py [hunk 1]
```diff
@@ -30,22 +30,61 @@
...
     for context, driver, keyword, arguments, parents in get_driver_options():
-        db[context].setdefault(driver, {'options': [], 'blocks': {}})
-        add_to = db[context][driver]
+        db.setdefault(context, {})
+        driver_options = {'options': [], 'blocks': {}}
+        for driver_alias in driver.split('/'):
+            db[context].setdefault(driver_alias, driver_options)
+        add_to = db[context][driver_alias]
         for parent in parents:
             add_to['blocks'].setdefault(parent, {'options': [], 'blocks': {}})
```
[the remark this answers is not in the document]
Comment (alltilla) [reply contrib/config_option_database/config_database.py]: `driver_options` and `{'options': [], 'blocks': {}}` in line 84 are different dicts.

I set `driver_options` to multiple drivers (`fifo` and `pipe`), so when I add entries to `driver_options`, both drivers get it.

In line 84 the dict is a new dict. If I used `driver_options` there, `add_to['blocks'][parent]` would use the driver's option dict, which I don't want.

File: contrib/config_option_database/config_database.py [hunk 2]
```diff
@@ -30,22 +30,61 @@
...
+def merge_block_as_option(keyword, arguments, options, blocks):
+    if keyword in blocks:
+        options.remove((keyword, arguments))
+        blocks[keyword]['options'].append(('', arguments))
+        return True
+    return False
+
+
+def merge_options_helper(current_dict):
+    options = current_dict['options']
```
[the remark this answers is not in the document]
Comment (alltilla) [reply contrib/config_option_database/config_database.py]: Renamed it and also did a little refactor for better readability.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by Kokan** [comment]
I plan to do the review by Friday, do not wait for me if it can be merged before I do the review.

**Comment by mitzkia** [comment]
I would like to propose to format differently options where values can be various types.
The main reason it can be confuse for eyes, it is a little bit similar to formatting blocks.

For example:
```console
  flags(
    <empty>
    <string>
  )
```

I would like to propose this format (pipe sign could means there can be various types):
```console
  flags( <empty> | <string> )
```

**Comment by alltilla** [comment]
> I would like to propose to format differently options where values can be various types.
> The main reason it can be confuse for eyes, it is a little bit similar to formatting blocks.
> 
> For example:
> 
> ```
>   flags(
>     <empty>
>     <string>
>   )
> ```
> 
> I would like to propose this format (pipe sign could means there can be various types):
> 
> ```
>   flags( <empty> | <string> )
> ```

Thanks! Done :)

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: contrib/config_option_database/config_database.py [hunk 3]
```diff
@@ -39,18 +39,67 @@ def parse_args():
...
+    _, arguments_list = option
+    return len(arguments_list) > 1
+
+
+def merge_mergeable_options(mergeable_options, options):
+    for keyword, arguments_list in mergeable_options:
+        merged_arguments = tuple()
+        for arguments in sorted(arguments_list):
+            if merged_arguments:
+                merged_arguments += ('|',)
```
Comment (Kokan) [inline contrib/config_option_database/config_database.py]: [optional]
My assumption upon checking the database(after the change):
* sequence is `"foo": [ <string> <number> ]`
* options is `"foo": [<string> | <number>]`

Before change:
* sequence is the same
* options is `"foo": [<string> ], "foo": [ <number>]`

Would it be possible not to change the database itself ? And provide the `|` in the view itself.
I like the new view, just not so much the database change.

Comment (gaborznagy) [reply contrib/config_option_database/config_database.py]: @mitzkia You don't need the database change just the view change, right?

Comment (alltilla) [reply contrib/config_option_database/config_database.py]: @kokan Yes, it is completely possible :)

Comment (Kokan) [reply contrib/config_option_database/config_database.py]: @gaborznagy  The view for a human is great, also the other changes are welcomed like listing.

Comment (alltilla) [reply contrib/config_option_database/config_database.py]: Pushed some new commits, the merging of options with multiple signatures is now only done at the printing stage.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**